### ISSUE TITLE
bad indexing in model segfaults

### ISSUE THREAD

**Issue by paciorek** [nimble-dev/nimble#601]
Perhaps more of a "very bad error trap" than a bug. And we do tell the user what is wrong to some degree. So not urgent.
 
```
  code <- nimbleCode( {
        for(i in 1:n)
            y[i] ~ dnorm(mu[n-i], 1)})
       m = nimbleModel(code, constants = list(n=3))
```

```
defining model...
Something is not the right size

 *** caught segfault ***
address (nil), cause 'memory not mapped'

Traceback:
 1: .Call(C_anyStochDependencies, graphExtPtr)
 2: nimbleGraph$anyStochDependencies()
 3: maps$setPositions3()
 4: md$setupModel(code = code, constants = constants, dimensions = dimensions,     userEnv = userEnv, debug = debug)
 5: BUGSmodel(code, name, constants, dimensions, data, inits, returnModel = !returnDef,     where, debug, check, calculate, userEnv)
 6: nimbleModel(code, constants = list(n = 3))
```

**Comment by perrydv** [comment]
In general for nimbleFunctions (including the nodeFunctions of a model), bad indexing can cause run-time crashes.

**Comment by perrydv** [comment]
Oh I see this is a different kind of issue than I initially thought.

**Comment by paciorek** [comment]
Addressed in PR #664.
